gdx/src/com/badlogic/gdx/scenes/scene2d/ui/Skin.java

```diff
@@ -440,6 +453,9 @@ public void setEnabled (Actor actor, boolean enabled) {
 			method.invoke(actor, style);
 		} catch (Exception ignored) {
 		}
+
+		Gdx.app.error("DEPRECATED",
+			"Setting style on Actor " + actor.getClass().getName() + ", which does not implement 'Styleable' interface");
```

Comment:
It's more of a warning than an error. I don't believe we do this kind of deprecated logging anywhere else.

We could deprecate `setEnabled(Actor, boolean)` and add `setEnabled(Styleable, boolean)`, which would have to cast to Actor. Callers would have to cast to remove ambiguity, eg `setEnabled((Styleable)moo, false)` which is a bummer. To improve that we could rename `setEnabled(Actor, boolean)` to `setEnabledReflection(Actor, boolean)` and deprecate it. Then we could use the fancy `<T extends Actor & Styleable> void setEnabled (T, boolean)`.

None of the options are amazing, but I think I like the logging the least.

Reply:
Thank you for the review!
I'm very indifferent about what deprecation method to use. I choose the logging, cause it avoided the ambigious method call problem, but I'm fine with the method overload too!

> To improve that we could rename setEnabled(Actor, boolean) to setEnabledReflection(Actor, boolean) and deprecate it.

Regarding this tho, I think this will break third-party libs that are compiled against an older libGDX version? However, if this method is rarely used anyway, it might be fine.

Reply:
Considering that migrating user code is simply to add `implements Styleable<MyWidget.Style>` to their custom widgets, for me it's a minor breaking change. So we could simply ditch the old code, no?

Reply:
I'm fully in favor of adding an `implements` clause to all my widgets in TextraTypist; it shouldn't be hard to do. It will mean that it will be briefly broken with nightly builds of libGDX and the next stable release until I can release TextraTypist with an update, so a stop-gap in the middle might help.

Reply:
Given there's not a super clean way to deprecate this one, I'm OK with a little breakage.

Reply:
Fine with breakage. Also fine with having deprecation migration hints as a fallback for a single release version, to be removed in next release.

Reply:
I implemented the breaking change with `setEnabledReflection`